geometry/optimization/hpolyhedron.cc

```diff
@@ -147,6 +148,46 @@ HPolyhedron HPolyhedron::Intersection(const HPolyhedron& other) const {
   return {A_intersect, b_intersect};
 }
 
+VectorXd HPolyhedron::UniformSample(
+    RandomGenerator* generator,
+    const Eigen::Ref<Eigen::VectorXd>& previous_sample) const {
+  std::uniform_real_distribution<double> uniform(-1, 1);
+  // Choose a random direction.
+  VectorXd direction(ambient_dimension());
+  for (int i = 0; i < direction.size(); ++i) {
+    direction[i] = uniform(*generator);
+  }
+  // Find max and min θ subject to
+  //   A(previous_sample + θ*direction) ≤ b.
+  VectorXd line_b = b_ - A_ * previous_sample;
+  VectorXd line_a = A_ * direction;
+  double theta_max = std::numeric_limits<double>::infinity();
+  double theta_min = -theta_max;
+  for (int i = 0; i < line_a.size(); ++i) {
+    if (line_a[i] < 0.0) {
+      theta_min = std::max(theta_min, line_b[i] / line_a[i]);
+    } else if (line_a[i] > 0.0) {
+      theta_max = std::min(theta_max, line_b[i] / line_a[i]);
+    }
+  }
+  if (std::isinf(theta_max) || std::isinf(theta_min)) {
```

Comment:
The math of finding [theta_min, theta_max] seems correct to me as `line_b` is always all non-negative elements and so`theta_min<0` and `theta_max<0`.